crd-generator/api-v2/src/main/java/io/fabric8/crdv2/generator/AbstractCustomResourceHandler.java

```diff
@@ -44,29 +46,35 @@ void addPrinterColumn(String path, String column, String format,
 
   protected void handlePrinterColumns(AbstractJsonSchema<?, ?> resolver, PrinterColumnHandler handler) {
     TreeMap<String, AnnotationMetadata> sortedCols = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    resolver.getAdditionalPrinterColumns().forEach(apc -> sortedCols.put(apc.path(), new AnnotationMetadata(apc, null)));
     sortedCols.putAll(resolver.getAllPaths(PrinterColumn.class));
     sortedCols.forEach((path, property) -> {
-      PrinterColumn printerColumn = ((PrinterColumn) property.annotation);
-      String column = printerColumn.name();
-      if (Utils.isNullOrEmpty(column)) {
-        column = path.substring(path.lastIndexOf(".") + 1).toUpperCase();
-      }
-      String format = printerColumn.format();
-      format = Utils.isNotNullOrEmpty(format) ? format : null;
-      int priority = printerColumn.priority();
-      String type = property.schema.getType();
-      if ("object".equals(type) || "array".equals(type)) {
-        LOGGER.warn("Printer column '{}' has a type '{}' that is not allowed, will use string intead", column, type);
-        type = "string";
-      } else if ("string".equals(type) && "date".equals(property.schema.getFormat())) {
-        type = "date";
-      }
+      if (property.annotation instanceof AdditionalPrinterColumn) {
+        AdditionalPrinterColumn printerColumn = ((AdditionalPrinterColumn) property.annotation);
+        String column = printerColumn.name();
+        String format = printerColumn.format() == Format.NONE ? null : printerColumn.format().getValue();
+        String type = printerColumn.type().getValue();
+        int priority = printerColumn.priority();
+        String description = printerColumn.getDescription();
+        handler.addPrinterColumn(path, column, format, priority, type, description);
+      } else {
+        PrinterColumn printerColumn = ((PrinterColumn) property.annotation);
+        String column = printerColumn.name();
+        String format = printerColumn.format();
+        format = Utils.isNotNullOrEmpty(format) ? format : null;
```

Comment:
We need those lines quite often.  Can we add a method to CRDUtils?

Something like:
```
 public static String emptyToNull(String s) {
    return Utils.isNullOrEmpty(s) ? null : s;
 }
```

Reply:
Yes, a utility method could be added in a follow-up.